hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/AzureNativeFileSystemStore.java

```diff
@@ -247,6 +249,8 @@
 
   private static final int DEFAULT_CONCURRENT_WRITES = 8;
 
+  private static final Charset METADATA_ENCODING = StandardCharsets.UTF_8;
```

Comment:
Is there a reason for using a different encoding when compared to the `AzureBlobFileSystemStore`?

Reply:
The documentation for Azure Data Lake Storage Gen2 (backing `AzureBlobFileSystemStore`) states that x-ms-properties must be encoded in ISO-8859-1 (see [path/update#request-headers](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update#request-headers)). The documentation for Azure Blob Storage (backing `AzureNativeFileSystemStore`) however only states that x-ms-meta must follow the conventions for C# identifiers (see [set-blob-metadata#request-headers](https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-metadata#request-headers)) which may contain Unicode letters and characters (see [identifier-names](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/identifier-names)). As such I believe that the two classes should use different encodings. Am I missing something?

Reply:
I was mostly curious about this one. I assume @DadanielZ  can confirm this one?

Reply:
This change is mostly looking good. If @DadanielZ  can confirm the encoding issue, i can happy to commit this one.

Reply:
Sorry for being away so long, yes, this is fine